core/src/main/java/io/hyperfoil/core/steps/ScheduleDelayStep.java

```diff
@@ -126,7 +126,7 @@ public Builder key(String key) {
       }
 
       public Builder duration(long duration, TimeUnit timeUnit) {
-         this.duration = timeUnit == null ? 0 : timeUnit.toMillis(duration);
+         this.duration = timeUnit == null ? 0 : timeUnit.toNanos(duration);
```

Comment:
we should check that is not negative - but TBH is a non-problem usually

Reply:
This check is already performed as part of the step's `build()`, e.g., https://github.com/Hyperfoil/Hyperfoil/blob/ae0cc9937ca9fa3a25ac737d07c99a5c567e407e/core/src/main/java/io/hyperfoil/core/steps/ScheduleDelayStep.java#L228

Reply:
yeah I see that, but I am expecting that when you call the `build()` on the `ScheduledDelayStep` it will be found that is negative here https://github.com/Hyperfoil/Hyperfoil/blob/ae0cc9937ca9fa3a25ac737d07c99a5c567e407e/core/src/main/java/io/hyperfoil/core/steps/ScheduleDelayStep.java#L224-L231

This is going to be called _after_ the duration is set (and converted)